lib/msprime.c

```diff
@@ -2967,17 +2994,31 @@ msp_run_single_sweep(msp_t *self, double max_time, unsigned long max_events)
     double *time = model->params.single_sweep.trajectory.time;
     double rec_rates[] = {0.0, 0.0};
     double sweep_pop_sizes[] = {0.0, 0.0};
-    double event_prob, event_rand, tmp_rand, e_sum;
+    double event_prob, event_rand, tmp_rand, e_sum, pop_size;
     double p_coal_b, p_coal_B, total_rate, sweep_pop_tot_rate;
     double p_rec_b, p_rec_B;
 
+    /* Keep the compiler happy */
+    sweep_pop_tot_rate = 0;
+    p_coal_b = 0;
+    p_coal_B = 0;
+
+    /* JK: I've removed the time and event limits on this function to simplify
+     * things as function is currently 'non-rentrant'; we can't stop in the middle
+     * of the sweep and pick it up again from where we left off later. This is
+     * different to all the other model runners. Probably the simplest thing to
+     * do here is to make the trajectory times absolute wrt to the simulation, and
+     * to then increment curr_step accordingly. We can then probably reason about
+     * when the call msp_single_sweep_initialise and msp_single_sweep_finalise
+     * depending on the value of curr_step, and hopefully reintroduce the max_time
```

Comment:
yes we would want to go back to the hudson coalescent after a given sweep to make guarantee the MRCA is reached

Reply:
> yes we would want to go back to the hudson coalescent after a given sweep to make guarantee the MRCA is reached

We'll do this with a SimulationModelChange at the top level --- we might not always want to change to Hudson, maybe we would want to finish the sim with DTWF, or SMC, or might just want to stop entirely.